dbms/src/Operators/Operator.h

```diff
@@ -101,9 +108,8 @@ class SourceOp : public Operator
 
     OperatorStatus awaitImpl() override { return OperatorStatus::HAS_OUTPUT; }
 };
-using SourceOpPtr = std::unique_ptr<SourceOp>;
+using SourceOpPtr = std::shared_ptr<SourceOp>;
```

Comment:
looks a bit hacky.
How about just holding the executor_id in `io_sources_map` and holding the connection_info in `DAGContext`?
`Map<String, ConnectionInfos> receiver_connection_infos_map`.